gflownet/utils/batch.py

```diff
@@ -0,0 +1,179 @@
+from collections import defaultdict
+from copy import deepcopy
+
+import numpy as np
+import torch
+
+from gflownet.utils.common import tbool, tfloat, tint, tlong
+
+
+class Batch:
+    """
+    Important note: one env should correspond to only one trajectory, all env_id should be unique
+    """
+
+    def __init__(self, loss, device, float):
+        self.loss = loss
+        self.device = device
+        self.float = float
+        self.envs = dict()
+        self.state = []
```

Comment:
Shouldn't all of these iterable variable names be plural? So states, actions, are_done, etc.

Reply:
yes, Alex fixed that in his branch with improvements to this PR